crates/matrix-sdk-ui/src/room_list_service/mod.rs

```diff
@@ -100,19 +102,29 @@ impl RoomListService {
     ///
     /// This won't start an encryption sync, and it's the user's responsibility
     /// to create one in this case using `EncryptionSync`.
-    pub async fn new(client: Client) -> Result<Self, Error> {
-        Self::new_internal(client, false).await
+    pub async fn new(
+        client: Client,
+        delay_before_recover: Option<Duration>,
```

Comment:
I'm not comfortable having `delay_before_recover` as an argument of the constructor. To me it feels like we need a builder.

Ideally, we would want to hardcode the value, and only expose a setter to change it when `#[cfg(test)]`. But since we want to change that via integration test, we would need a `testing` feature, which is something we don't want 🔁.

So, I _think_ we should move the specific tests (about this delay) inside this crate: they become a unit tests and we can use the `#[cfg(test)] fn set_delay_before_recover(…)` approach.

What do you think? It would remove the need for this argument in the constructor, it keeps the code as simple as before, we don't need to make it dynamic, it's a hardcoded value that can be overwritten for tests only. Thoughts?

Reply:
That is a lot simpler indeed :) 

Done !